ext/standard/string.c

```diff
@@ -1080,7 +1080,7 @@ PHP_FUNCTION(strtok)
 
 	if (!BG(strtok_string)) {
 		/* String to tokenize not set. */
-		// TODO: Should this warn?
+		php_error_docref(NULL, E_WARNING, "First call must set the string argument to tokenize with `%s`", ZSTR_VAL(tok));
```

Comment:
I'm fine with either error messages... as long as no string user input is exposed (`ZSTR_VAL(tok)`). :)

Just my 2c:
`Both arguments must be provided when calling strtok() the first time`

I'm really not sure which version is the best :)

Reply:
Ok, the "first time" isn't really appropriate because the parameter must be provided whenever there is a new tokenization starting, so: `Both arguments must be provided when starting tokenization`

Reply:
oky doky.